woocommerce/woocommerce-ios
# Fix memory leaks from view controllers - product form, Aztec, search, order details

Fixes #2069 

## Background

As @shiki noticed in #2069, the `KeyboardFrameObserver` had a strong reference to the view controller that owns it and thus the view controller couldn't be deallocated after leaving the screen. I audited the memory state with Xcode's memory debugger after navigating to a few screens and found a few lingering view controllers. This PR fixes the leaking view controllers in these screens.

## Changes

- Product form and related view controllers from navigation:
  - `ProductFormViewController`
  - `ProductPriceSettingsViewController`
  - `ProductInventorySettingsViewController`
  - `AztecEditorViewController`
    - `TextViewViewController`
    - `AztecLinkFormatBarCommand`: weak'ed the reference to the view controller that presents the link settings form
- Order details:
  - `ManualTrackingViewController`
  - `ShipmentProvidersViewController`
- Search:
  - `SearchViewController`
  - `OrderSearchStarterViewController`

## Testing

We can verify by navigating to the following screens and leaving them back to the root for all tabs, and then check that none of the view controllers that we navigated to exist in the memory via Xcode memory debugger.

Prerequisite: the site has shipment tracking plugin activated 

- Launch the app
- Edit product:
  - Go to the Products tab
  - Tap on a simple Product
  - Tap into the product name to edit the text
  - Tap into the product description to the Aztec editor
  - In the Aztec editor, select some text and tap the link icon on the toolbar to insert a link to the text
  - Go back to the edit product main screen
  - Tap on the price settings row and edit the price, then go back to the edit product main screen
  - Tap on the inventory settings row and edit the sku, then go back to the edit product main screen
  - Go back to the Products tab
- Search:
  - Go to the Orders tab
  - Tap on the search icon in the navigation bar, make sure the keyboard is up
  - Type some text that renders no results
  - Go back to the Orders tab
- Order details:
  - Go to the Orders tab
  - Tap on a processing order
  - Tap "Add Tracking" row under "OPTIONAL TRACKING INFORMATION" section
  - Tap the "Tracking number" row to enter some text
  - Tap the "Shipping carrier" row
  - Tap into the search bar to search for some carriers
  - Leave the manual tracking screens and go back to the Orders tab
- Run the memory debugger in Xcode --> only the root view controllers of the 4 tabs (Dashboard/stats, Orders, Products, Reviews) should be present in the memory like in the screenshot below

<img width="443" alt="Screen Shot 2020-04-08 at 9 34 28 AM" src="https://user-images.githubusercontent.com/1945542/78739770-8eb51780-7987-11ea-8d5c-3e2d4b28b35f.png">


Update release notes:
